File: 917-BoatsToSavePeople/917-BoatsToSavePeople.java
// Last updated: 7/31/2025, 5:19:30 PM
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int l=0;
        int r=people.length-1;
        int boats=0;
        while(l<=r){
            if(people[l]+people[r]<=limit){
                l++;
            }
            r--;
            boats++;
        }
        return boats;
    }
}

File: 917-BoatsToSavePeople/917-BoatsToSavePeople.py
# Last updated: 7/31/2025, 5:19:36 PM
class Solution(object):
    def numRescueBoats(self, people, limit):
        people.sort()
        b=0
        l,r=0,len(people)-1
        while l<=r:
            if people[l]+people[r]<=limit:
                l+=1
            r-=1
            b+=1
        return b
        """
        :type people: List[int]
        :type limit: int
        :rtype: int
        """
        